src/core/reference/include/openvino/reference/interpolate.hpp

```diff
@@ -444,18 +454,21 @@ void InterpolateEval<T>::linear_onnx_func(const T* input_data, T* out) {
     // or
     //     num_of_axes == input_rank - 2.
     // Hence, if num_of_axes != input_rank, then interpolated axes indices are
-    //    [0, 1, ..., num_of_axes - 1]
-    // Otherwise, if num_of_axes == input_rank, interpolated axes indices are
     //    [2, 3, ..., num_of_axes - 1]
+    // Otherwise, if num_of_axes == input_rank, interpolated axes indices are
+    //    [0, 1, ..., num_of_axes - 1]
     const int64_t axis_idx_offset = (input_rank == num_of_axes) ? 2 : 0;
 
     const int64_t spatial_rank = info.spatial_rank;
     const int64_t points_in_neighbor = 1LL << spatial_rank;
 
     const T* xdata = input_data;
     T* ydata = out;
+
     for (int64_t n = 0; n < batch_size; ++n) {
         for (int64_t c = 0; c < num_channels; ++c) {
+            if (channel_last && c > 0)
```

Comment:
Can this checks be remove from loop?
The algorithm is known when `channel_last` flag value is set. Some functions or policy can be used to apply actions in modified steps of algorithm.